packages/expo/src/effects/NativeLinearGradient.web.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+
+type Props = {
+  colors: number[];
+  locations?: number[] | null;
+  startPoint?: Point | null;
+  endPoint?: Point | null;
+  onLayout?: Function;
+} & React.ElementProps<View>;
+
+type State = {
+  width: number;
+  height: number;
+};
+type Point = [number, number];
+
+export default class NativeLinearGradient extends React.PureComponent<Props, State> {
+  state = {
+    width: 0.5,
```

Comment:
Does 0.5 have any significance?

Also might be better to default these to null and to render no backgroundImage until the layout is calculated. Generally with UIs we get two chances to show content: the initial render (ex: blank cell or placeholder) and the final render (actual content). Once we show something that looks like content to the user, it will draw their attention -> if the initial render looks too much like real content but is substantially different from the final content then the jarring transition happens when the user is focused on that UI element, making it worse (ex: different gradient angle or different font -- for font loading, rendering unstyled text and then applying a custom font a second later is worse that just waiting that extra second to show anything at all). In this case, if we don't know the gradient angle then it's probably better not to show an incorrect guess for one frame.